artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -1460,23 +1473,47 @@ private PagedReference moveNext() {
 
       private PagedReference internalGetNext() {
          for (;;) {
+            if (logger.isDebugEnabled() && currentPageIterator == null) {
```

Comment:
Is the null value unlikely to happen? Seems like it since it will presumably NPE below if so. (EDIT: so do we even need the logging?)

If its the rarer event, I'd put its check first. I'd probably just make it 2 ifs.